06-building-a-program-start-to-finish/solving-the-problem/Output/invertedIndex.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Inverted index class
+ */
+class InvertedIndex {
+  /**
+   * Inverted index constructor
+   */
+  constructor() {
+    // Object to hold the index
+    this.index = {};
+  }
+
+  /**
+   * strips words of all special characters and numbers
+   * and returns an Array of each words
+   * @param{String} words - String to tokenize
+   * @return{Array} list of words devoid of special characters or symbols
+   */
+  static tokenize(words) {
+    return words.trim().replace(/-/g, ' ')
+      .replace(/[^A-z\s]/g, '')
+      .toLowerCase()
+      .split(' ')
+      .sort();
+  }
+
+  /**
+   * remove duplicate words in tokenize array
+   * @param{String} words - The string to be filtered
+   * @return{Array} tokens - Without duplicated words
+   */
+  static uniqueWords(words) {
+    const tokens = InvertedIndex.tokenize(words);
+    return tokens.filter((item, index) => tokens.indexOf(item) === index);
+  }
+
+  /**
+   * creates an inverted index
+   * takes a file name and array of words as
+   * argument and creates inverted index
+   * @param{String} fileName - The name of the file to be indexed
+   * @param{Array} fileToIndex - Array of contents of the JSON file to index
+   * @return{String} for invalid files - JSON file is Empty
+   * @return{Object || String} for valid files returns
+   * index - That maps words to locations(documents)
+   * while for empty files returns JSON file is empty
+   */
+  createIndex(fileName, fileToIndex) {
+    const wordsToIndex = [];
+    const fileIndex = {};
+    const fileLength = fileToIndex.length;
+    if (fileLength === 0) {
+      return 'JSON file is Empty';
+    }
+    fileToIndex.forEach((document) => {
+      if (document.text && document.title) {
+        wordsToIndex
+          .push(`${document.text
+            .toLowerCase()}`);
+      }
+    });
+    const uniqueContent = InvertedIndex.uniqueWords(wordsToIndex.join(' '));
+    uniqueContent.forEach((word) => {
+      fileIndex[word] = [];
+      wordsToIndex.forEach((document, indexPosition) => {
+        if (document.indexOf(word) > -1) {
+          fileIndex[word].push(indexPosition);
+        }
+      });
+    });
+    this.index[fileName] = fileIndex;
```

Comment:
no, this just creates the index and getIndex gets the index [getters and setters]